<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qMRMLMarkupsCurveSettingsWidget</class>
 <widget class="QWidget" name="qMRMLMarkupsCurveSettingsWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>680</width>
    <height>647</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="ctkCollapsibleButton" name="curveSettingsCollapseButton" native="true">
     <property name="text" stdset="0">
      <string>Curve Settings</string>
     </property>
     <property name="collapsed" stdset="0">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QWidget" name="curveSettingsWidget" native="true">
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <property name="leftMargin">
          <number>9</number>
         </property>
         <property name="topMargin">
          <number>9</number>
         </property>
         <property name="rightMargin">
          <number>9</number>
         </property>
         <property name="bottomMargin">
          <number>9</number>
         </property>
         <item>
          <layout class="QFormLayout" name="formLayout_3">
           <item row="0" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Curve Type:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="curveTypeComboBox"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Constrain to Model:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="qMRMLNodeComboBox" name="modelNodeSelector" native="true">
             <property name="nodeTypes" stdset="0">
              <stringlist>
               <string>vtkMRMLModelNode</string>
              </stringlist>
             </property>
             <property name="noneEnabled" stdset="0">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="ctkCollapsibleGroupBox" name="surfaceCurveCollapsibleButton">
           <property name="title">
            <string>Surface</string>
           </property>
           <property name="collapsed" stdset="0">
            <bool>false</bool>
           </property>
           <layout class="QFormLayout" name="formLayout_2">
            <item row="0" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Cost function:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Distance weighting function:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QWidget" name="scalarFunctionWidget" native="true">
              <property name="styleSheet">
               <string notr="true"/>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <property name="spacing">
                <number>0</number>
               </property>
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QLabel" name="scalarFunctionPrefixLabel">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>distance / (</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="scalarFunctionLineEdit">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="sizeIncrement">
                  <size>
                   <width>0</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="toolTip">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;justify&quot;&gt;Function used to calculate scalar weights for pathfinding. The currently active point scalar can be accessed using the &amp;quot;activeScalar&amp;quot; variable name. All other scalars can be accessed as variables provided that they don't contain any illegal characters (&amp;quot;./* etc.).&lt;/p&gt;&lt;p&gt;Example functions: &amp;quot;activeScalar*activeScalar&amp;quot;, &amp;quot;exp(activeScalar)&amp;quot;, &amp;quot;1.0-activeScalar&amp;quot;, &amp;quot;scalarA*scalarB&amp;quot;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                 <property name="text">
                  <string>activeScalar</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="scalarFunctionSuffixLabel">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>)^2</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="costFunctionComboBox"/>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="ctkCollapsibleGroupBox" name="CurveSettingsAdvancedOptionCollapsibleGroupBox">
           <property name="title">
            <string>Advanced</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
           <property name="collapsed" stdset="0">
            <bool>true</bool>
           </property>
           <property name="collapsedHeight" stdset="0">
            <number>7</number>
           </property>
           <layout class="QGridLayout" name="CurveSettingsAdvancedOptionGridLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="projectCurveMaxSearchRadiusLabel">
              <property name="text">
               <string>Maximum projection distance:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="ctkSliderWidget" name="projectCurveMaxSearchRadiusSliderWidget" native="true">
              <property name="toolTip">
               <string>Select the maximum projection distance as percentage of bounding box size.</string>
              </property>
              <property name="singleStep" stdset="0">
               <double>0.100000000000000</double>
              </property>
              <property name="minimum" stdset="0">
               <double>0.000000000000000</double>
              </property>
              <property name="maximum" stdset="0">
               <double>100.000000000000000</double>
              </property>
              <property name="value" stdset="0">
               <double>25.000000000000000</double>
              </property>
              <property name="suffix" stdset="0">
               <string> %</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="resampleCurveCollapsibleButton" native="true">
     <property name="toolTip">
      <string>Resample a curve with the number of points specified.</string>
     </property>
     <property name="text" stdset="0">
      <string>Resample</string>
     </property>
     <property name="collapsed" stdset="0">
      <bool>true</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout1">
      <item row="0" column="0">
       <widget class="QLabel" name="resampleCurveOutputNodeLabel">
        <property name="text">
         <string>Output node: </string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="resampleCurveNumerOfOutputPointsLabel">
        <property name="toolTip">
         <string>Select the number of points on the resampled curve. </string>
        </property>
        <property name="text">
         <string>Number of resampled points: </string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="ctkDoubleSpinBox" name="resampleCurveNumerOfOutputPointsSpinBox" native="true">
        <property name="focusPolicy">
         <enum>Qt::TabFocus</enum>
        </property>
        <property name="toolTip">
         <string>The active curve will be resamples with the number of points specified.</string>
        </property>
        <property name="decimals" stdset="0">
         <number>0</number>
        </property>
        <property name="minimum" stdset="0">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum" stdset="0">
         <double>1000000.000000000000000</double>
        </property>
        <property name="singleStep" stdset="0">
         <double>1.000000000000000</double>
        </property>
        <property name="value" stdset="0">
         <double>20.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="resampleCurveOutputNodeSelector" native="true">
        <property name="toolTip">
         <string>Select a node to store the resampled curve </string>
        </property>
        <property name="nodeTypes" stdset="0">
         <stringlist>
          <string>vtkMRMLMarkupsCurveNode</string>
         </stringlist>
        </property>
        <property name="noneEnabled" stdset="0">
         <bool>true</bool>
        </property>
        <property name="addEnabled" stdset="0">
         <bool>true</bool>
        </property>
        <property name="removeEnabled" stdset="0">
         <bool>true</bool>
        </property>
        <property name="renameEnabled" stdset="0">
         <bool>true</bool>
        </property>
        <property name="selectNodeUponCreation" stdset="0">
         <bool>true</bool>
        </property>
        <property name="noneDisplay" stdset="0">
         <string>(Overwrite current node)</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="3">
       <widget class="QPushButton" name="resampleCurveButton">
        <property name="toolTip">
         <string>Resamples the active curve with the number of points specified.</string>
        </property>
        <property name="text">
         <string>Resample curve</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>ctkCollapsibleGroupBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkSliderWidget</class>
   <extends>QWidget</extends>
   <header>ctkSliderWidget.h</header>
  </customwidget>
  <customwidget>
   <class>ctkDoubleSpinBox</class>
   <extends>QWidget</extends>
   <header>ctkDoubleSpinBox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
 </connections>
</ui>
